src/shared/exec-util.c

```diff
@@ -365,8 +366,47 @@ static int gather_environment_consume(int fd, void *arg) {
         return r;
 }
 
+ExecCommandFlags exec_command_flags_from_strv(char **ex_opts) {
+        ExecCommandFlags flags = 0, ex_flag = 0;
+        char **opt = NULL;
+
+        STRV_FOREACH(opt, ex_opts) {
+                if (!isempty(*opt)) {
+                        ex_flag = exec_command_flags_from_string(*opt);
+                        if (ex_flag != (ExecCommandFlags) -1)
+                                flags |= ex_flag;
+                }
+        }
+
+        return flags;
+}
+
+char **exec_command_flags_to_strv(ExecCommandFlags flags) {
+        ExecCommandFlags it = flags;
+        char **ex_opts = NULL;
+        int i;
+
+        for (i = 0; it != 0; it &= ~(1 << i), i++) {
+                if (FLAGS_SET(flags, (1 << i)))
+                        if (strv_extend(&ex_opts, exec_command_flags_to_string((1 << i))) < 0)
+                                log_oom();
+        }
+
+        return ex_opts;
+}
+
 const gather_stdout_callback_t gather_environment[] = {
         gather_environment_generate,
         gather_environment_collect,
         gather_environment_consume,
 };
+
+static const char* const exec_command_flags_table[] = {
+        [EXEC_COMMAND_IGNORE_FAILURE] = "ignore-failure",
+        [EXEC_COMMAND_FULLY_PRIVILEGED] = "privileged",
+        [EXEC_COMMAND_NO_SETUID] = "no-setuid",
+        [EXEC_COMMAND_AMBIENT_MAGIC] = "ambient",
+        [EXEC_COMMAND_NO_ENV_EXPAND] = "no-env-expand",
```

Comment:
uh, this is not OK. this table looks like it only has 5 entries, but actually has 2^5 entries, 32 entries of which 28 are not used. While wasting 28 pointers might not be that bad for now as soon as we define a few more bits like this the waste will grow exponentially.

This needs to be solved differently really, so that there's no table indexed by bitmask, but by the bit index if you so will